src/BizHawk.Client.Common/movie/tasproj/ZwinderStateManager.cs

```diff
@@ -311,6 +311,49 @@ public void EvictReserved(int frame)
 			}
 		}
 
+		public void Unreserve(int frame)
+		{
+			// Before removing the state, check if it should be still be reserved.
+			// For now, this just means checking if we need to keep this state to satisfy the ancient interval.
+			if (ShouldKeepForAncient(frame))
+				return;
+
+			EvictReserved(frame);
+		}
+
+		private void PossiblyMoveToReserved(ZwinderBuffer buffer, int stateIndex)
+		{
+			var state = buffer.GetState(stateIndex);
+
+			// Add to reserved buffer if externally reserved, or if it matches an "ancient" state consideration
+			if (_reserveCallback(state.Frame) || ShouldKeepForAncient(state.Frame))
+				AddToReserved(state);
+			else
+				StateCache.Remove(state.Frame);
+		}
+
+		/// <summary>
+		/// Will removing the state on this leave us with a gap larger than the ancient interval?
+		/// </summary>
+		private bool ShouldKeepForAncient(int frame)
+		{
+			int index = StateCache.BinarySearch(frame + 1);
+			if (index < 0)
+				index = ~index;
+			if (index == StateCache.Count)
+			{
+				// There is no future state... so why are we wanting to evict this state?
+				// We aren't decaying from _recent, that's for sure. So,
+				return false;

```

Comment:
I don't really understand this comment and code. Assuming there is no next state, shouldn't the return value be `true`, aka always keep the most current state unconditionally?

Reply:
Both this and the below case can absolutely be hit because `ShouldKeepForAncient` is called from `Unreserve` which is the public api used to unreserve states and could be called with any frame really.

> I figured false was safer

How so? false means allowing the state to be removed. I would imagine keeping it should be the safe option, especially when this is the state on the highest frame.

Reply:
Looking at it again I guess it doesn't really matter. I'd still like different comments to clarify that calls from `Unreserve` could hit this code path and why the return value is what it is (same for the `index <= 1` below).